trpcultivate_phenotypes/src/Plugin/Validators/DuplicateTraits.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/**
+ * @file
+ * Contains validator plugin definition.
+ */
+
+namespace Drupal\trpcultivate_phenotypes\Plugin\Validators;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesTraitsService;
+use Drupal\trpcultivate_phenotypes\TripalCultivatePhenotypesValidatorBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Validate duplicate traits within a file
+ *
+ * @TripalCultivatePhenotypesValidator(
+ *   id = "trpcultivate_phenotypes_validator_duplicate_traits",
+ *   validator_name = @Translation("Duplicate Traits Validator"),
+ *   validator_scope = "FILE ROW",
+ * )
+ */
+class DuplicateTraits extends TripalCultivatePhenotypesValidatorBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * A nested array of already validated values forming the unique trait name +
+   *   method name + unit combinations that have been encountered by this
+   *   validator so far within the input file
+   */
+  protected $unique_traits = [];
+
+  /**
+   * Traits Service
+   */
+  protected $service_traits;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, TripalCultivatePhenotypesTraitsService $service_traits) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->service_traits = $service_traits;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition){
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('trpcultivate_phenotypes.traits')
+    );
+  }
+
+  /**
+   * Validate the values within the cells of this row.
+   * @param array $row_values
```

Comment:
There should always be an empty line after the single line description and before the parameters.

```suggestion
   *
   * @param array $row_values
```